Grammar (#6132)
FuelLabs/sway: docs/book/src/index.md

Rust shines because it can use zero-cost abstractions and its sophisticated borrow-checker memory model to achieve impressive runtime performance for complex programs without a garbage collector.

> On a blockchain, cost of execution and deployment is scarce.

Reviewer: This suggestion doesn't really make sense either. The cost isn't scarce, it's just costly.

```suggestion
On a blockchain, execution and deployment are limited by transaction costs.
```

Reviewer 2: There may be a better term than cost to analogize with space and time, which are resources.

> Memory usage is low and execution time is short.

Reviewer: ```suggestion
Memory is an expensive resource.
```

Reviewer 2: You're losing key information and this isn't what I mean here.

Author: Updated
